android/quest/src/main/java/org/smartregister/fhircore/quest/util/PatientUtil.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.util
+
+import ca.uhn.fhir.rest.gclient.ReferenceClientParam
+import ca.uhn.fhir.rest.gclient.TokenClientParam
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.Search
+import com.google.android.fhir.search.search
+import org.hl7.fhir.r4.model.CodeableConcept
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Condition
+import org.hl7.fhir.r4.model.Enumerations
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.ResourceType
+import org.smartregister.fhircore.engine.configuration.ConfigurationRegistry
+import org.smartregister.fhircore.quest.configuration.view.Code
+import org.smartregister.fhircore.quest.configuration.view.DynamicColor
+import org.smartregister.fhircore.quest.configuration.view.Filter
+import org.smartregister.fhircore.quest.configuration.view.PatientRegisterRowViewConfiguration
+import org.smartregister.fhircore.quest.configuration.view.Properties
+import org.smartregister.fhircore.quest.configuration.view.Property
+import org.smartregister.fhircore.quest.data.patient.model.AdditionalData
+
+suspend fun loadAdditionalData(
+  patientId: String,
+  configurationRegistry: ConfigurationRegistry,
+  fhirEngine: FhirEngine
+): List<AdditionalData> {
+  val result = mutableListOf<AdditionalData>()
+
+  val patientRegisterRowViewConfiguration =
+    configurationRegistry.retrieveConfiguration<PatientRegisterRowViewConfiguration>(
+      configClassification = QuestConfigClassification.PATIENT_REGISTER_ROW
+    )
+
+  patientRegisterRowViewConfiguration.filters?.forEach { filter ->
+    when (filter.resourceType) {
```

Comment:
Handle all the branches of the when expression on `enum`. You can add an `else` branch instead and return an empty list or `null` to handle the rest of the missing branches.

Reply:
I've used the `if` statement in place of `when`  as we have only one condition for now.